### ISSUE TITLE
Integrate MLflow Logging

### ISSUE THREAD

**Issue by unniishere** [roboflow/rf-detr#110]
It will be good if you can integrate Mlflow logging like you guys did for wandb and tensorboard as wandb is paid.

**Comment by robmarkcole** [comment]
Plus 1

**Comment by lab176344** [comment]
PR is open https://github.com/roboflow/rf-detr/pull/109

**Comment by isaacrob-roboflow** [comment]
what is the benefit of mlflow vs tensorboard?

**Comment by robmarkcole** [comment]
It is a server based solution, widely used by teams since it can be self hosted

**Comment by isaacrob-roboflow** [comment]
neat! 

one thing I'm worried about is supporting an unlimited number of logging modules. would be convenient if we'd built on top of pytorch lightning and we'd get all these for free but 🤷 shipped on VERY short schedule.

@SkalskiP do you have thoughts on shipping too many logging frameworks? it does seem like there's a good argument for this one

**Comment by isaacrob-roboflow** [comment]
we should make sure to keep these learnings in mind for the next model we ship

**Comment by robmarkcole** [comment]
I've had conversations with lightning about logger support - they basically support log metric and log_dict as standard. It gets logger specific for artefact logging. Overall +1 for using lightning for the training loop - this was a factor that held me back from embracing ultralytics

**Comment by isaacrob-roboflow** [comment]
yeah I am a really big fan of pytorch lightning, just didn't make sense to refactor for this project and probably too late now

**Comment by lab176344** [comment]
@isaacrob-roboflow we host mlflow servers in gcp, has almost the same functionalities as W&B. Ultralytics has support for Mlflow too.  +1 for lighting too =)

**Comment by ggand0** [comment]
+1

**Comment by Akhp888** [comment]
+1

**Comment by robmarkcole** [comment]
@isaacrob-roboflow @SkalskiP the PR is ready, could really use this functionality :-)

**Comment by ArunAniyan** [comment]
+1

**Comment by Sid-Pawar-13-TML** [comment]
+1

**Comment by a-shahrestani** [comment]
+1

**Comment by isaacrob** [comment]
@Borda thoughts?

**Comment by Borda** [comment]
We will use lightning which has it integrated

**Comment by Borda** [comment]
Hey @unniishere, great news — this was shipped in PR #109! MLflow logging is now available alongside WandB and TensorBoard.

You can enable it by passing `logger="mlflow"` (or the equivalent config) to `model.train()`. Please try it out on the latest release (`pip install -U rfdetr`) and let us know if anything isn't working as expected. If you run into issues, feel free to reopen.

Thanks for the request!
